cmd/incus-migrate/main_migrate.go

```diff
@@ -537,7 +554,7 @@ func (c *cmdMigrate) Run(cmd *cobra.Command, args []string) error {
 	// Automatically clean-up the temporary path on exit
 	defer func(path string) {
 		_ = unix.Unmount(path, unix.MNT_DETACH)
-		_ = os.Remove(path)
+		_ = os.RemoveAll(path)
```

Comment:
The root.img file is still left behind. I think it might still be mounted when the temp directory is cleaned up.

Reply:
I'd definitely avoid `RemoveAll` as for containers if the unmount fails somehow, we may find ourselves deleting a lot of actual data...

Reply:
I don't know what that Unmount is doing here too, I don't see us ever making that path a mount.

I'd still avoid `RemoveAll` though as for containers we do mount stuff in sub-paths.
If that doesn't go away correctly, we may find ourselves recursively deleting a lot of data with RemoveAll.

We should know what files to expect in there so we can specifically delete those.

Reply:
OK, switched back to just `Remove` and I added the potentially converted image as another path to remove.

Do we want to make an issue about the temp directory itself not being properly cleaned up?

Reply:
Do you have a list of whats' left in there?

Reply:
```
gibmat@futurfusion:/tmp$ sudo ls -ltr /tmp/incus-migrate_mount_1349025577/
total 0
-rw-r--r-- 1 root root 0 Sep 13 08:34 root.img
```

Reply:
Okay, so we should go ahead and delete that one prior to attempting to delete the directory.